### PR TITLE
input: doc: document event definitions

### PR DESCRIPTION
**Pull request by kartben**
Added missing documentation for event types, key codes, etc.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]
great to see Doxygen getting some love

**Review by nashif — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Comment by kartben** [comment]
@fabiobaltieri please confirm I got everything right :) In particular:
* confirm that LT/RT are left trigger and right trigger
* *_RX and *_RY is for rotations
* INPUT_MSC_SCAN: really wasn't sure with this one, it looks like it's only being used for testing, so I even wonder if it should not be part of this file?

**Comment by kartben** [comment]
> @fabiobaltieri please confirm I got everything right :) In particular:
> 
> * confirm that LT/RT are left trigger and right trigger
> * *_RX and *_RY is for rotations
> * INPUT_MSC_SCAN: really wasn't sure with this one, it looks like it's only being used for testing, so I even wonder if it should not be part of this file?

oops, posted after seeing you had already +1'd :)

**Comment by fabiobaltieri** [comment]
> > @fabiobaltieri please confirm I got everything right :) In particular:
> > 
> > * confirm that LT/RT are left trigger and right trigger
> > * *_RX and *_RY is for rotations
> > * INPUT_MSC_SCAN: really wasn't sure with this one, it looks like it's only being used for testing, so I even wonder if it should not be part of this file?
> 
> oops, posted after seeing you had already +1'd :)

To be honest there's no hard definition of these, they are picked from https://lxr.missinglinkelectronics.com/linux/include/uapi/linux/input-event-codes.h, I technically cannot import the file straight on but a bit a time most of those will get through.

Scan is indeed scan code, RX RY and RZ, I *think* they are actually meant to be the same as X Y Z but for the right stick, so maybe worth changing it.

**Review by fabiobaltieri — CHANGES_REQUESTED** [review]

**Comment by fabiobaltieri** [comment]
I'll be totally honest, I'm not mad for documenting these since most are either self explanatory (`KEY_A`), awkward to explain (the whole south-north-west-east story, there's no button called like that usually, it's just the typical 4 button gamepad to avoid the A-B X-Y vs B-A Y-X ambiguity), or bound to interpretation by the application. That said, if you think it's good practice let's do it. The two Linux file we can refer to are https://lxr.missinglinkelectronics.com/linux/include/uapi/linux/input-event-codes.h and https://lxr.missinglinkelectronics.com/linux/drivers/hid/hid-debug.c

File: include/zephyr/dt-bindings/input/input-event-codes.h [hunk 1]
```diff
@@ -22,119 +22,120 @@
...
  */
-#define INPUT_ABS_RX 0x03
-#define INPUT_ABS_RY 0x04
-#define INPUT_ABS_RZ 0x05
-#define INPUT_ABS_X 0x00
-#define INPUT_ABS_Y 0x01
-#define INPUT_ABS_Z 0x02
+#define INPUT_ABS_RX 0x03               /**< Absolute rotation around X axis */
+#define INPUT_ABS_RY 0x04               /**< Absolute rotation around Y axis */
+#define INPUT_ABS_RZ 0x05               /**< Absolute rotation around Z axis */
```
[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply include/zephyr/dt-bindings/input/input-event-codes.h:114]: Well, after some chit chat research it turns that is mentioned in the hid tables and explicitly says rotation, let's go with it then.

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**